policy-controller/k8s/status/src/http_route.rs

```diff
@@ -29,64 +30,60 @@ pub enum InvalidParentReference {
     SpecifiesSection,
 }
 
-pub(crate) fn make_parents(http_route: policy::HttpRoute) -> Result<Vec<ParentReference>> {
-    let namespace = http_route
-        .metadata
-        .namespace
-        .expect("HTTPRoute must have a namespace");
-    let parents = ParentReference::collect_from(http_route.spec.inner.parent_refs, &namespace)?;
-    Ok(parents)
+/// Convenience trait to get shared route reference types.
+pub(crate) trait GetReferences {
```

Comment:
I don't think this trait does anything for us right now.  I understand the goal is to set up for when there are different Route types, but I don't think this makes sense right now and may not even be how we want to handle this in the future.

Reply:
I'm to blame for the trait :/ After re-working a bunch of stuff, turns out the additional indirection doesn't really simplify much anyway, it's pretty trivial to get these references when we don't error out from the function. We could go back to the way things were but instead of accepting a route and requiring a clone, we can just work on a route ref.

```rust
fn make_parents(route_ref: &HttpRoute) ...
```

Reply:
Yep sounds good I'm good with removing it.